src/operator/bufferTime.ts

```diff
@@ -168,13 +168,13 @@ class BufferTimeSubscriber<T> extends Subscriber<T> {
     this.contexts = null;
   }
 
-  protected onBufferFull(context: Context<T>) {
+  protected onBufferFull(context: Context<T>): void {
     this.closeContext(context);
     const closeAction = context.closeAction;
     closeAction.unsubscribe();
     this.remove(closeAction);
 
-    if (this.timespanOnly) {
+    if (this.timespanOnly && this.contexts !== null) {
```

Comment:
Not sure. This occurs with async context, not able to think trivial way to create test cases at the moment writing PR. Let me think a bit more.